script.js

```diff
@@ -1,4 +1,125 @@
+  return statementOneOuputValue;
+};
+
+var statmentTwo = function (playerTurn, chosenDice, beatItNum) {
+  if (playerTurn < maxPlayers) {
+    var statementTwo = `Welcome player ${playerTurn}, you chose dice ${chosenDice} first.` + '<br>' + `Your number is ${beatItNum}` + '<br>' + 'It is now player 2\'s turn';
+  } else {
+    var statementTwo = `Welcome player ${playerTurn}, you chose dice ${chosenDice} first.` + '<br>' + `Your number is ${beatItNum}`;
+  }
+  return statementTwo;
+};
+
+var statementThree = function () {
+  var statementThree = [];
+  var arrayLength = playerScore.length;
+  for (var i = 0; i < arrayLength; i++) {
+    var player = Number([i]) + 1;
+    statementThree.push(`player ${player}'s number ${playerScore[i]}` + '<br>');
+  }
+  return statementThree;
+};
+/* ------------------DICE FUNCTIONS---------------------------- */
+var rollDice = function (min, max) {
+  // generate a random number from min to max
+  var diceRollOutput = Math.floor(Math.random() * (max - min + 1)) + min;
+  return diceRollOutput;
+};
+
+var rollDiceMode = function () {
+  var counter = 0;
+  // roll 2 dice
+  while (counter < 2) {
+    playerDiceRoll.push(rollDice(1, 6));
+    counter += 1;
+  }
+  return playerDiceRoll;
+};
+
+var showDiceMode = function (playerTurn) {
+  var diceOne = playerDiceRoll[0];
+  var diceTwo = playerDiceRoll[1];
+  var showDiceModeOuput = statementOne(playerTurn, diceOne, diceTwo);
+  return showDiceModeOuput;
+};
+
+/* ------------------USER CHOICE FUNCTIONS---------------------------- */
+var pickDiceMode = function (playerTurn, playerInput) {
+  var returnStatment = '';
+  var playerChoice = playerInput - 1;
+  var index = 0;
+  while (index < playerDiceRoll.length) {
+    var currentRoll = playerDiceRoll[index];
+    if (playerChoice == index) {
+      playerReorder.push(currentRoll);
+      playerDiceRoll.splice(playerChoice, 1);
+    }
+    index += 1;
+  }
+  // add remaining number to Reordered array
+  playerReorder.push(playerDiceRoll[0]);
+  // store player number
+  var playerBeatItNum = Number(`${playerReorder[0]}${playerReorder[1]}`);
+  playerScore.push(playerBeatItNum);
+  // return statment
+  returnStatment = statmentTwo(playerTurn, playerInput, playerBeatItNum);
+  // purge
+  playerDiceRoll = [];
+  playerReorder = [];
+  return returnStatment;
+};
+
+/* ----------------------------------------------------------- */
+/* ------------------MAIN FUNCTION---------------------------- */
+/* ----------------------------------------------------------- */
 var main = function (input) {
-  var myOutputValue = 'hello world';
+  // Call input playerInput to make game logic clearer
+  var playerInput = input;
+  var myOutputValue = '';
+  if (gameMode == ROLL_DICE_MODE && playerTurnSwitch <= maxPlayers) {
+    rollDiceMode();
+    gameMode = SHOW_DICE_MODE;
+  }
+  if (gameMode == SHOW_DICE_MODE && playerTurnSwitch <= maxPlayers) {
```

Comment:
Does this need to be a separate conditional from the previous conditional? I see that this conditional will always run after the previous conditional, and there is no `return` statement in between